src/MarketV2.sol

```diff
@@ -0,0 +1,679 @@
+        }
+        uint credit = getCreditLimitInternal(borrower);
+        debtManager.increaseDebt(borrower, amount);
+        require(credit >= debtManager.debt(borrower), "Exceeded credit limit");
+        dola.transfer(to, amount);
+        emit Borrow(borrower, amount);
+    }
+
+    /**
+    @notice Function for borrowing DOLA.
+    @dev Will borrow to msg.sender
+    @param amount The amount of DOLA to be borrowed.
+    */
+    function borrow(uint amount) public {
+        borrowInternal(msg.sender, msg.sender, amount);
+    }
+
+    /**
+    @notice Function for using a signed message to borrow on behalf of an address owning an escrow with collateral.
+    @dev Signed messaged can be invalidated by incrementing the nonce. Will always borrow to the msg.sender.
+    @param from The address of the user being borrowed from
+    @param amount The amount to be borrowed
+    @param deadline Timestamp after which the signed message will be invalid
+    @param v The v param of the ECDSA signature
+    @param r The r param of the ECDSA signature
+    @param s The s param of the ECDSA signature
+    */
+    function borrowOnBehalf(address from, uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) public {
+        require(deadline >= block.timestamp, "DEADLINE_EXPIRED");
+        unchecked {
+            address recoveredAddress = ecrecover(
+                keccak256(
+                    abi.encodePacked(
+                        "\x19\x01",
+                        DOMAIN_SEPARATOR(),
+                        keccak256(
+                            abi.encode(
+                                keccak256(
+                                    "BorrowOnBehalf(address caller,address from,uint256 amount,uint256 nonce,uint256 deadline)"
+                                ),
+                                msg.sender,
+                                from,
+                                amount,
+                                nonces[from]++,
+                                deadline
+                            )
+                        )
+                    )
+                ),
+                v,
+                r,
+                s
+            );
+            require(recoveredAddress != address(0) && recoveredAddress == from, "INVALID_SIGNER");
+            borrowInternal(from, msg.sender, amount);
+        }
+    }
+
+    /**
+    @notice Internal function for withdrawing from the escrow
+    @dev The internal function is shared by the withdraw function and withdrawOnBehalf function
+    @param from The address owning the escrow to withdraw from.
+    @param to The address receiving the tokens
+    @param amount The amount being withdrawn.
+    */
+    function withdrawInternal(address from, address to, uint amount) internal {
+        uint limit = getWithdrawalLimitInternal(from);
+        require(limit >= amount, "Insufficient withdrawal limit");
+        require(dbr.deficitOf(from) == 0, "Can't withdraw with DBR deficit");
```

Comment:
Does this rule still apply? I think it should be replaced with the user's debtManager's deficit instead, no?